### PR TITLE
Add spawn reason and properly implement teleport cause

### PR DESCRIPTION
**Pull request by Snow-Pyon**
Target Minecraft versions: any.
Requirements: none.
Related issues: #1444

Description:
Adds two new classinfos with their desired event values, one being the requested spawn reason and the other being the teleport cause which was not implemented properly (it didn't have a classinfo registered).

I mainly made the PR to get input on the spawn reason's names (the ones from the english.lang).

### CODE REVIEW

File: src/main/resources/lang/english.lang [hunk 1]
```diff
@@ -1197,6 +1197,37 @@ inventory types:
...
+    jockey: jockey
+    lightning: lightning
+    mount: mount
+    natural: natural
+    nether_portal: nether portal
+    ocelot_baby: ocelot baby
+    reinforcements: reinforcements
+    shoulder_entity: perching, shoulder
+    silverfish_block: silverfish reveal, silverfish trap
+    slime_split: slime_split
```
Comment (Blueyescat) [inline src/main/resources/lang/english.lang]: _

**Comment by Blueyescat** [comment]
I think you should also add support for spawn reason and teleport cause to [CondDamageCause](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/conditions/CondDamageCause.java).

**Comment by Snow-Pyon** [comment]
@Blueyescat I thought the same thing though I didn't really see the need of it due to two things:
1. We don't really want more conditions.
2. No one has ever used that condition (I only saw two persons using it in four years).

File: src/main/resources/lang/english.lang (hunk 1 above)
Comment (Snow-Pyon) [reply src/main/resources/lang/english.lang]: whoops

File: src/main/resources/lang/english.lang [hunk 2]
```diff
@@ -1242,8 +1273,11 @@ types:
 	damagecause: damage cause¦s @a
 	teleportcause: teleport cause¦s @a
 	inventoryaction: inventory action¦s @a
-	clickaction: click actions¦s @a
+	clicktype: click type¦s @a
```
Comment (TheLimeGlass) [inline src/main/resources/lang/english.lang]: click[ed] type

click type, clicked type

Comment (Snow-Pyon) [reply src/main/resources/lang/english.lang]: Nouns are handled differently and thus they don't allow this.

**Comment by bensku** [comment]
No time for review now, but concerning point 7; *yes*, we are generally not adding German translations unless someone who speaks it contributes them.

**Comment by Snow-Pyon** [comment]
1. It could be good but there's something that we would need to think before adding it and that is: is this worth it? What I am saying here is that if it isn't going to be used a lot, it will be just wasted space in memory since we would need to save that data ourselves. This isn't something we need to care about for damage cause since MC/Spigot stores that data.

I could possibly add a config option for it disabled by default, and if it is enabled by many users I could consider fully implementing it but that's about it. The other way around would be request it to the Paper/Spigot guys and see if they implement it.

2. I can do it after (or perhaps before) merging this PR, yeah.

3. This is actually a pretty good idea, but since the main goal of this PR is adding the spawn reason event value, I'll just do it in another PR and see what other Skript/Addon developers think about it.

4. I deleted it thinking that there was no specific use for it though I see it now, going to implement it for both. Though, I really think this kind of expressions is pretty pointless, a way to implement this without having to register a new expression could be good.

5. I don't quite understand this, you want me to add a teleport cause <-> location comparator? I don't really think it makes sense if that is the case. The damage causes have a damage cause <-> item type comparator since some damage causes have the same alias as some items, which might also be the case of teleport cause.

6. I'll remove it later.

7. Bensku already answered this.

**Comment by TheLimeGlass** [comment]
5 needs some way to get the location from and to. Event values only return one and past and present event values don't work as explained in another old issue. So there needs to be an implemented way of getting from and to.

**Comment by TheLimeGlass** [comment]
@bensku don't you think we should add language support? More users come from the French language than English.

**Comment by Snow-Pyon** [comment]
4. I know but that isn't exactly what I was talking about. I was more about the `[the] %*classinfo%` kind of event value, there are many classes with this format which is fine but it is just boilerplate code, hence why we should add a way to do this without needing to register a new expression for it (perhaps making `event-` optional in ExprEventExpression would do the job? It would have to be tested)

5. Ah, you were talking about locations. Past and future event values need to be fixed before we add this, a workaround for it (i.e, a separate expression) isn't an option.

And about the language support, this isn't totally related to this PR so you're better off creating an issue about it, or just say your thoughts on #1359 (we're actually thinking about removing language support at all too).

**Comment by Pikachu920** [comment]
someone who knows french would have to be willing to maintain them before that's even considered. plus, language support in skript is rather meh in the first place... what's the point of language files when the syntaxes are English anyway

**Comment by TheLimeGlass** [comment]
There could be methods like `acceptChange and change` and if a syntax has it, it would allow the syntax to have been accepted in that language that would be returned.

Would be a better feature when the parser becomes asynchronous though.

**Comment by jaylawl** [comment]
@bensku im a native german speaker - what's the scope of what needs to be translated?

**Pull request merged**